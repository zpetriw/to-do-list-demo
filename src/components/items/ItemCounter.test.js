import React from 'react';
import expect from 'expect';
import {shallow} from 'enzyme';
import configureMockStore from 'redux-mock-store';
import ConnectedItemCounter, {ItemCounter} from './ItemCounter';
import ItemCounterDisplay from './ItemCounterDisplay';
import AboutPage from '../about/AboutPage';

describe('When ItemCounter renders', () => {

    describe('its bare component', () => {
    
        let wrapper;
    
        beforeEach( () => {
            wrapper = shallow(<ItemCounter/>);
        });
    
        it('should render ItemCounterDisplay', () => {
            expect(wrapper.find(ItemCounterDisplay).length).toBe(1);
        });
    
    });
    
    describe('its connect(mapStateToProps) component', () => {
    
        let wrapper;
        let store;
        let mockStore;
    
        beforeEach(() => {
            mockStore = configureMockStore();
            store = mockStore({
                items: ["item1", "item2"]
            });
            // You need to dive or else you will look at Connect(ItemCounter) and not see any children.
            wrapper = shallow(<ConnectedItemCounter store={store}/>).dive();
        });
    
        it('should render ItemCounterDisplay', () => {
            expect(wrapper.find(ItemCounterDisplay).length).toBe(1);
        });
    
        it('should render ItemCounterDisplay with a correct itemCount prop', () => {
            expect(wrapper.find(ItemCounterDisplay).props().itemCount).toBe(2);
        });
    });
});



