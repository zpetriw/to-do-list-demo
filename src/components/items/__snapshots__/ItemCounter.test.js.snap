// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When ItemCounter renders its bare component should render correctly 1`] = `
<li
  className="MuiListItem-root-2 MuiListItem-default-5 MuiListItem-gutters-9"
  disabled={false}
>
  <div
    className="MuiListItemText-root-12 Component-counter-1"
  >
    <h3
      className="MuiTypography-root-18 MuiTypography-subheading-25 MuiListItemText-primary-15"
    >
      Items To Do: 1
    </h3>
  </div>
</li>
`;
